### PR TITLE
Add an Android bindings project

### PR DESCRIPTION
**Pull request by hwsmm**
- Supersedes https://github.com/ppy/SDL2-CS/pull/180
- Required by https://github.com/ppy/osu-framework/pull/6105

This PR adds an Android bindings project to the repository, and a workflow to build binaries for Android.
I separated it from the original project since building it requires Xamarin, and I thought that most consumers of this project won't like it.

Workflow may need to be run once to replace my binaries. Couldn't test workflow since I couldn't run in my forked repository, probably due to secrets?

### CODE REVIEW

**Comment by hwsmm** [comment]
I just removed them.

**Comment by smoogipoo** [comment]
Lgtm. I bundled one additional change here for a header that's been removed upstream.

**Review by smoogipoo — APPROVED** [review]
Let's do this!

File: SDL3-CS.sln [hunk 1]
```diff
@@ -6,27 +6,26 @@ MinimumVisualStudioVersion = 10.0.40219.1
...
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "SDL3-CS", "SDL3-CS\SDL3-CS.csproj", "{3AEE5979-6974-4BD6-9BB1-1409B0BB469B}"
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "SDL3-CS", "SDL3-CS\SDL3-CS.csproj", "{3AEE5979-6974-4BD6-9BB1-1409B0BB469B}"
+EndProject
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "SDL3-CS.SourceGeneration", "SDL3-CS.SourceGeneration\SDL3-CS.SourceGeneration.csproj", "{432C86D0-D371-4D01-BFFE-01D2CDCCA7B8}"
 EndProject
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "SDL3-CS.SourceGeneration", "SDL3-CS.SourceGeneration\SDL3-CS.SourceGeneration.csproj", "{432C86D0-D371-4D01-BFFE-01D2CDCCA7B8}"
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "SDL3-CS.Tests", "SDL3-CS.Tests\SDL3-CS.Tests.csproj", "{CF980481-8227-4BED-970E-6433C83F64CB}"
 EndProject
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "SDL3-CS.Tests", "SDL3-CS.Tests\SDL3-CS.Tests.csproj", "{CF980481-8227-4BED-970E-6433C83F64CB}"
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "SDL3-CS-Android", "SDL3-CS-Android\SDL3-CS-Android.csproj", "{CA28F49C-D0BE-47D6-9E82-7A0B8C380B8B}"
```
Comment (Susko3) [inline SDL3-CS.sln:22]: Hmm, we'd usually name this `SDL3-CS.Android`, not `-Android`

Comment (smoogipoo) [reply SDL3-CS.sln:22]: Yep, I've got fixes on the way...

**Pull request merged**